tools/unstableWarningAnonymizer/unstableAnonScript.chpl

```diff
@@ -31,84 +31,187 @@ features.
-import SortedMap.sortedMap as map;
+import Map.map as map;
+// import SortedMap.sortedMap as map;
+import Set.set;
+import Sort;
 import Regex.regex;
 import ArgumentParser;
 
-proc countUniqueWarnings(ref warningsMap: map(string, int), inputFileReader: IO.fileReader(?)){
+proc countUniqueWarnings(ref warningsMap: map(string, ?t), inputFileReader: IO.fileReader(?)) where t == (int, set(string)){
   // A pattern of a typical warning line
   // Ex: filename.chpl:lineNumber: warning message blah
-  const warningRegex = new regex(".*.chpl:\\d+: (.*)\n"); // The  (.*) is a capture group
+  const warningRegex = new regex("(.*.chpl):\\d+: (.*)\n"); // Anything inside ( ) is a capture group
   var warning : string;
-  for (fullMatch, warningMatch) in inputFileReader.matches(warningRegex, captures=1) {
+  var fileName: string;
+  for (fullMatch, fileNameMatch, warningMatch) in inputFileReader.matches(warningRegex, captures=2) {
     inputFileReader.extractMatch(warningMatch, warning);
+    inputFileReader.extractMatch(fileNameMatch, fileName);
     // Check if the string mentions that something is unstable
     // If so, add it to the map
-    if warning.find("unstable") == -1 then
+    if !containsUnstableWarning(warning) then
       continue;
+    warning = anonmizeWarning(warning);
     if warningsMap.contains(warning) {
-      warningsMap[warning] += 1;
+      warningsMap[warning][0] += 1;
     } else {
-      warningsMap[warning] = 1;
+      warningsMap[warning][0] = 1;
     }
+    warningsMap[warning][1].add(fileName); // Add the fileName to the set of files that have this warning
   }
 }
 
+proc containsUnstableWarning(warning: string) : bool {
+  // Check if the string mentions that something is unstable
+  return warning.find("unstable") != -1 || warning.find("enum-to-bool") != -1 ||
+         warning.find("enum-to-float") != -1;
+}
+
 
-inline proc prettyPrintMap(m : map(?), writer: IO.fileWriter(?)){
-  for key in m.keys() {
-    if m[key] < 2 then
-      writer.writeln(m[key], " instance of \"", key, "\"");
-    else writer.writeln(m[key], " instances of \"", key, "\"");
+proc anonmizeWarning(warning: string) : string {
+  // Anonymize known warning messages that include variable names
+  // when so that it doens't reveal variable names or other impl detauls
+
+  const forallRef ="warning: inferring a 'ref' intent on an array in a forall is unstable - in the future this may require an explicit 'ref' forall intent for";
+  if warning.find(forallRef) != -1 then
+    return forallRef + " <var-name>";
+
+  const typeName = "warning: using a type's name ";
+  const typeNameUse = "in a 'use' statement to access its tertiary methods is an unstable feature";
+  const typeNameImport = "in an 'import' statement to access its tertiary methods is an unstable feature";
+  if warning.find(typeName) != -1 && warning.find(typeNameUse) != -1 then
+    return typeName + typeNameUse;
+  if warning.find(typeName) != -1 && warning.find(typeNameImport) != -1 then
```

Comment:
Not essential, but you could probably structure this so that the `typeName` check only happens once.  E.g.
```
if warning.find(typeName) != 1 {
  if warning.find(typeNameUse) != -1 then
    return typeName + typeNameUse;
  else if warning.find(typeNameImport) != -1 then
    return typeName + typeNameImport;
}
```